api/src/pcapi/scripts/clean_offer_titles_with_eans/main.py

```diff
@@ -0,0 +1,242 @@
+from dataclasses import dataclass
+from datetime import datetime
+from datetime import timezone as tz
+import functools
+import json
+import logging
+from typing import Callable
+from typing import Collection
+from typing import cast
+
+import sqlalchemy as sa
+
+from pcapi.core.bookings import api as bookings_api
+from pcapi.core.mails import transactional as transactional_mails
+from pcapi.core.offers.models import GcuCompatibilityType
+from pcapi.core.offers.models import Offer
+from pcapi.core.offers.models import OfferValidationStatus
+from pcapi.flask_app import app
+from pcapi.models import db
+from pcapi.models.offer_mixin import OfferValidationType
+from pcapi.repository import atomic
+from pcapi.repository import on_commit
+from pcapi.utils.chunks import get_chunks
+
+
+logger = logging.getLogger(__name__)
+
+# Mandatory since this module uses atomic() which needs an application context.
+app.app_context().push()
+
+
+BOOKS_CDS_VINYLES_QUERY = """
+    SELECT
+        sub.id,
+        sub.ean,
+        sub.name,
+        sub."subcategoryId",
+        sub."isActive",
+        p.id is not null as exists,
+        p.id as product_id,
+        p.name as product_name,
+        p."jsonData" as product_json_data,
+        p."gcuCompatibilityType"
+    FROM (
+        SELECT
+            id,
+            substring("name" similar '%#"[[:digit:]]{13}#"%' escape '#') as ean,
+            name,
+            "subcategoryId",
+            "isActive"
+        FROM
+            offer
+        WHERE
+            "name" similar to '%\\d{13}%'
```

Comment:
Sur Metabase cette requête ne renvoie rien. Par contre, si je fais ce changement, alors ça marche : 
```suggestion
            "name" similar to '%[0-9]{13}%'
```

Reply:
La syntaxe de _Metabase_ n'est peut-être pas exactement la même que celle de PG.
J'ai revérifié en local et en **staging** et la requête s'exécute bien.

⚠️ le double `\\` est nécessaire dans le contexte SQLA, pour lancer la requête directement dans un _shell_ _psql_, il faut n'en laisser qu'un.